### PR TITLE
feature/#497 extract business logic from repo to entity (p2)

### PR DESCRIPTION
**Pull request by toan-quach**
#497 
Only extracting business logic of:
- [x] datanode

Notes:
The lack of new tests is due to what we are expecting from this ticket. The behaviors should all work the same after the refactoring. Hence, the current tests are more than sufficient to make sure that happens!

### CODE REVIEW

**Review by jrobinAV — APPROVED** [review]

File: src/taipy/core/data/data_node.py [hunk 1]
```diff
@@ -534,3 +536,97 @@ def all_subclasses(cls):
...
+                elif isinstance(properties[exposed_type_key], List):
+                    properties[exposed_type_key] = [
+                        v if v in valid_str_exposed_types else f"{v.__module__}.{v.__qualname__}"
+                        for v in properties[exposed_type_key]
+                    ]
+                else:
+                    properties[exposed_type_key] = (
+                        f"{properties[exposed_type_key].__module__}." f"{properties[exposed_type_key].__qualname__}"
+                    )
+        return properties
```
Comment (jrobinAV) [inline src/taipy/core/data/data_node.py:558]: The deserialize_exposed_type method is specific for tabular data nodes only. It should not be in data_node class.

File: src/taipy/core/data/data_node.py [hunk 2]
```diff
@@ -534,3 +536,97 @@ def all_subclasses(cls):
...
+                elif isinstance(properties[exposed_type_key], dict):
+                    properties[exposed_type_key] = {
+                        k: v if v in valid_str_exposed_types else locate(v)
+                        for k, v in properties[exposed_type_key].items()
+                    }
+                elif isinstance(properties[exposed_type_key], List):
+                    properties[exposed_type_key] = [
+                        v if v in valid_str_exposed_types else locate(v) for v in properties[exposed_type_key]
+                    ]
+        return properties
```
Comment (jrobinAV) [inline src/taipy/core/data/data_node.py:575]: Same remark as the serialize_exposed_type method.

File: src/taipy/core/data/abstract_sql.py [hunk 3]
```diff
@@ -232,3 +232,19 @@ def __setattr__(self, key: str, value) -> None:
...
+        )
+        return properties
+
+    @classmethod
+    def _deserialize_datanode_properties(cls, data_node_model):
+        properties = super()._deserialize_datanode_properties(data_node_model)
+        properties = super()._deserialize_exposed_type(
+            properties, cls.__EXPOSED_TYPE_PROPERTY, cls.__VALID_STRING_EXPOSED_TYPES
+        )
+
```
Comment (jrobinAV) [inline src/taipy/core/data/abstract_sql.py:249]: That could be the opportunity to put these methods in a TabularDataNode class for factorization.

https://app.zenhub.com/workspaces/taipy-613ef328affde6000ff77fb3/issues/gh/avaiga/taipy-core/350

File: src/taipy/core/data/data_node.py [hunk 4]
```diff
@@ -534,3 +536,97 @@ def all_subclasses(cls):
...
+
+    def _serialize_datanode_properties(self) -> dict:
+        properties = self._properties.data.copy()
+        return properties
+
+    @classmethod
+    def _deserialize_datanode_properties(cls, data_node_model) -> dict:
+        properties = data_node_model.data_node_properties.copy()
+        return properties
```
Comment (jrobinAV) [inline src/taipy/core/data/data_node.py:584]: I don't see a lot of value in having concrete methods for _serialize_datanode_properties and _deserialize_datanode_properties in the data_node class. It is just one line factorized.

I would make these 2 methods abstract in data node class.

**Review by jrobinAV — COMMENTED** [review]
Your implementation seems to work, but it reveals the need of factorizing some code for tabular data nodes. 

We can benefit from this current ticket to implement the other ticket https://app.zenhub.com/workspaces/taipy-613ef328affde6000ff77fb3/issues/gh/avaiga/taipy-core/350

What do you think?

File: src/taipy/core/data/abstract_sql.py (hunk 3 above)
Comment (toan-quach) [reply src/taipy/core/data/abstract_sql.py:249]: well sure why not :D should we implement it the way we did with _AbstractFileDataNode?

File: src/taipy/core/data/data_node.py (hunk 4 above)
Comment (toan-quach) [reply src/taipy/core/data/data_node.py:584]: oh initially, only generic, json and json dn has an implementation of these functions, while csv, parquet, excel don't. So I keep it as a concrete class so I don't have to recreate them in classes that didn't need it. But it's not the case at the end of the PR :D I'll fix this

**Pull request merged**